plugin/storage/es/factory.go

```diff
@@ -71,45 +79,22 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		return err
 	}
 	f.primaryClient = primaryClient
-	// TODO init archive (cf. https://github.com/jaegertracing/jaeger/pull/604)
+	archiveClient, err := f.archiveConfig.NewClient(logger, metricsFactory)
+	if err != nil {
+		return err
+	}
+	f.archiveClient = archiveClient
 	return nil
 }
 
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
-	cfg := f.primaryConfig
-	return esSpanStore.NewSpanReader(esSpanStore.SpanReaderParams{
-		Client:            f.primaryClient,
-		Logger:            f.logger,
-		MetricsFactory:    f.metricsFactory,
-		MaxSpanAge:        cfg.GetMaxSpanAge(),
-		MaxNumSpans:       cfg.GetMaxNumSpans(),
-		IndexPrefix:       cfg.GetIndexPrefix(),
-		TagDotReplacement: cfg.GetTagDotReplacement(),
-	}), nil
+	return createSpanReader(f.metricsFactory, f.logger, f.primaryClient, f.primaryConfig, false)
 }
 
 // CreateSpanWriter implements storage.Factory
 func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
-	cfg := f.primaryConfig
-	var tags []string
-	if cfg.GetTagsFilePath() != "" {
-		var err error
-		if tags, err = loadTagsFromFile(cfg.GetTagsFilePath()); err != nil {
-			f.logger.Error("Could not open file with tags", zap.Error(err))
-			return nil, err
-		}
-	}
-	return esSpanStore.NewSpanWriter(esSpanStore.SpanWriterParams{Client: f.primaryClient,
-		Logger:            f.logger,
-		MetricsFactory:    f.metricsFactory,
-		NumShards:         f.primaryConfig.GetNumShards(),
-		NumReplicas:       f.primaryConfig.GetNumReplicas(),
-		IndexPrefix:       f.primaryConfig.GetIndexPrefix(),
-		AllTagsAsFields:   f.primaryConfig.GetAllTagsAsFields(),
-		TagKeysAsFields:   tags,
-		TagDotReplacement: f.primaryConfig.GetTagDotReplacement(),
-	}), nil
+	return createSpanWriter(f.metricsFactory, f.logger, f.archiveClient, f.primaryConfig, false)
```

Comment:
@pavolloffay I think the client here should be `f.primaryClient` and not `f.archiveClient`